spec/repositories/stream_daily_averages_repository_spec.rb

```diff
@@ -21,4 +21,33 @@
       )
     end
   end
+
+  describe '#from_time_range' do
+    it 'returns daily averages for given stream from time range' do
+      stream = create_stream!
+      stream_daily_average1 =
+        create_stream_daily_average!({ stream: stream, date: 1.day.ago })
+      stream_daily_average2 =
+        create_stream_daily_average!({ stream: stream, date: 2.days.ago })
+      old_stream_daily_average =
+        create_stream_daily_average!({ stream: stream, date: 5.months.ago })
+      create_stream_daily_average!
+
+      result1 = subject.from_time_range(stream_id: stream.id, start_date: 3.days.ago, end_date: 1.day.ago)
+      result2 = subject.from_time_range(stream_id: stream.id, start_date: 3.days.ago, end_date: 2.days.ago)
+      result3 = subject.from_time_range(stream_id: stream.id, start_date: 6.months.ago, end_date: 1.day.ago)
```

Comment:
I think it's enough to have one check here. In general, I would go with an approach that in one test case (so within `it` block) a tested method (`from_time_range` in this case) is called once. 

If there is something more to test, for example some edge cases, I would add them as separate test cases and add `context` block if needed.